src/SitemapGenerator.php

```diff
@@ -116,7 +133,25 @@ public function getSitemap(): Sitemap
      */
     public function writeToFile(string $path)
     {
-        $this->getSitemap()->writeToFile($path);
+        $sitemap = $this->getSitemap();
+
+        if ($this->chunk) {
+            // Call the sitemap generation and process each created sitemap
+            $index = SitemapIndex::create();
+            $format = preg_replace('/\.xml/', '_%d.xml', $path);
+            $this->sitemaps->each(function (Sitemap $sitemap, int $key) use ($index, $format) {
+                $path = sprintf($format, $key);
+
+                $sitemap->writeToFile(sprintf($format, $key));
+                $index->add(last(explode('public', $path)));
+            });
+
+            $index->writeToFile($path);
+        }
+
+        else {
```

Comment:
I very much dislike else statements. Let's avoid it by moving ` $index->writeToFile($path);` about of the `if` block.